tailnet/coordinator.go

```diff
@@ -48,6 +49,17 @@ type Coordinator interface {
 	ServeMultiAgent(id uuid.UUID) MultiAgentConn
 }
 
+// CoordinatorV2 is the interface for interacting with the coordinator via the 2.0 tailnet API.
+type CoordinatorV2 interface {
+	// ServeHTTPDebug serves a debug webpage that shows the internal state of
+	// the coordinator.
+	ServeHTTPDebug(w http.ResponseWriter, r *http.Request)
+	// Node returns an in-memory node by ID.
+	Node(id uuid.UUID) *Node
+	Close() error
+	Coordinate(ctx context.Context, id uuid.UUID, name string, a TunnelAuth) (chan<- *proto.CoordinateRequest, <-chan *proto.CoordinateResponse)
```

Comment:
Curious on your thoughts: on public interfaces I normally like returning a wrapper with functions to access the channels instead of returning raw channels. Does that feel too abstracty for this? I could feel it being slightly verbose needing to always store two channels.

Reply:
It's a good instinct, and I think it depends on how we expect it to be used. In this case, I'm imagining that we will have an object that "glues" the Coordinate() call to the drpc Stream.

e.g.

```
type coordinateStreamer struct {
    stream proto.DRPCStream_Coordinate
    reqs chan<- *proto.CoordinateRequests
    resps <-chan *proto.CoordinateResponse
}
```

And this thing will have very simple methods that copy from the channels into the stream and vice versa.

Thus, I'm kinda already planning to wrap them up in a struct as soon as they are returned, so putting them together in a wrapper just means I have to unwrap.

Reply:
Makes sense :+1: